### DISCUSSION TITLE
beta3 useMutation TContext Question

### DISCUSSION THREAD

**Discussion by dusty** [TanStack/query#1201] [Q&A] [answered]
I'm playing with beta3 and trying to make a custom hook, and I'm getting a type error that I need to provide 4 types to useMutation options.  I can't figure out what the TContext type is supposed to be.

<img width="808" alt="Screen Shot 2020-10-21 at 17 37 47 " src="https://user-images.githubusercontent.com/792/96804591-478ef100-13c4-11eb-8b9a-edc61791f15f.png">

Here is an example
```
export const useSignupMutation = (
  options?: UseMutationOptions<SignupMutation, GraphqlError, SignupMutationVariables>
) => {
  return useMutation<SignupMutation, GraphqlError, SignupMutationVariables>(
    graphqlFetchFn<SignupMutation, SignupMutationVariables>(SignupDocument),
    options
  )
}
```

**Comment by boschni** [comment, marked as answer]
It's the value returned from the `onMutate` callback: https://react-query-beta.tanstack.com/guides/optimistic-updates . But the `UseMutationOptions` generics should have defaults so I added them in the latest beta.

**Reply by dusty** [reply]
Thanks!
